File: FixHTML.java
import java.io.*;
import java.nio.file.*;
import java.util.regex.*;

public class FixHTML {
    public static void main(String[] args) throws Exception {
        String[] files = {"copilot.html", "mock_interview.html", "resume.html", "records.html"};
        String dir = "H:\\java spring\\interviewai\\src\\main\\resources\\templates\\";
        
        Pattern pattern = Pattern.compile("\\?+/([a-zA-Z0-9]+)>");
        
        for (String filename : files) {
            File f = new File(dir + filename);
            if (!f.exists()) continue;
            
            String content = new String(Files.readAllBytes(f.toPath()), "UTF-8");
            Matcher matcher = pattern.matcher(content);
            String repaired = matcher.replaceAll("</$1>");
            
            Files.write(f.toPath(), repaired.getBytes("UTF-8"));
            System.out.println("Repaired " + filename);
        }
    }
}

File: fix_html.py
import os
import re

files = ['copilot.html', 'mock_interview.html', 'resume.html', 'records.html']
directory = r'H:\java spring\interviewai\src\main\resources\templates'

for filename in files:
    filepath = os.path.join(directory, filename)
    with open(filepath, 'r', encoding='utf-8', errors='ignore') as f:
        content = f.read()

    # Repair broken tags (e.g. ??/span> or ?/title> -> </span> or </title>)
    repaired = re.sub(r'\?+?/([a-zA-Z0-9]+)>', r'</\1>', content)
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(repaired)

print("Repair completed.")
